### PR TITLE
Add Edit Message Functionality

### LINKED ISSUE #74: Edit Message Feature
As an EmbedChat developer

**I need to**:  Add an option to edit a message if the message belongs to that user.

**So That**: EmbeddedChat Users can edit their own messages.

**Acceptance Criteria**

- [ ] Implement the Editing Message functionality and the messages should be updated in realtime.

### PR DESCRIPTION
**Pull request by ayush3160**
# Implemented Editing message functionailty.

## Acceptance Criteria fulfillment

- [x] Implement the Editing Message functionality and the messages should be updated in realtime.

Fixes #74 

## Video/Screenshots

https://user-images.githubusercontent.com/89914602/210564367-b80b4456-0782-4792-87e8-ec75ed62e242.mp4

### CODE REVIEW

**Comment by ayush3160** [comment]
@sidmohanty11 , Can you please review this PR.

Thanks.

File: src/index.js [hunk 1]
```diff
@@ -26,6 +26,12 @@ export const RCComponent = ({
   const setToastbarPosition = useToastStore((state) => state.setPosition);
   setToastbarPosition(toastBarPosition);
 
+  const [messageToEdit, setMessageToEdit] = useState({});
```
Comment (sidmohanty11) [inline src/index.js:29]: Let's not have any prop-drilling or global state that can affect rendering of components. As we are using zustand as our global state management can we please stick to it?

**Review by sidmohanty11 — CHANGES_REQUESTED** [review]
Hey @ayush3160 thanks a lot for the PR! The feature looks amazing!

**Comment by ayush3160** [comment]
@sidmohanty11 , Please have a look I have used zustand for managing the state.

File: src/components/ChatInput/ChatInput.js [hunk 2]
```diff
@@ -8,14 +8,16 @@ import styles from './ChatInput.module.css';
 import { EmojiPicker } from '../EmojiPicker/index';
 import RCContext from '../../context/RCInstance';
 import { useGoogleLogin } from '../../hooks/useGoogleLogin';
-import { useToastStore, useUserStore } from '../../store';
+import { useToastStore, useUserStore, useEditMessageStore } from '../../store';
```
Comment (sidmohanty11) [inline src/components/ChatInput/ChatInput.js:11]: I think it will be better to use the useMessageStore rather than creating a new store

**Review by sidmohanty11 — CHANGES_REQUESTED** [review]

**Comment by ayush3160** [comment]
@sidmohanty11 , Please have a look I have used the message store

**Pull request merged**